pkg/pushsync/pushsync.go

```diff
@@ -134,13 +138,17 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 	defer streamer.Close()
 
 	wc, rc := protobuf.NewWriterAndReader(streamer)
-
 	if err := ps.sendChunkDelivery(wc, chunk); err != nil {
 		return fmt.Errorf("forward chunk to peer %s: %w", peer.String(), err)
 	}
 	receiptRTTTimer := time.Now()
 
-	receipt, err := ps.receiveReceipt(rc)
+	// Most of the times now you wont get a tag because
+	// /bytes and /files API does not implement tags
+	// SO dont print any logs or returning error, if you dont find them
+	t, _ := ps.tagger.Get(chunk.TagID())
```

Comment:
you only need  to call this once in handler.
be consistent. call it before `sendChunkDelivery`
and pass `t` to both `sendChunkDelivery` and `receiveReceipt`

Reply:
SendChunkDelivery is called from 2 places. one from this handler and the other from pusher. We have 2 choise

1) implemet this in handler and in pusher
2) implement it her and in SendChunkDelivery

if you chose the second method, then all the things related to tag stays in one package. Thats why i chose this method.

Reply:
forwarded chunks do not have a tag, and therefore checking within the function or even passing  it  is incorrect.
Sorry all  I missed this, so no, this handler should have at no place do anything from this 
In fact this points to the solution being that not only tag calls should also be directly within the PushToClosest function body only (which is called from the pusher), the increments  of StateSent and StateSynced should come right after error checking `sendChunkDelivery` and the `setChunkSynced` calls, respectively.

Reply:
the above you described which was the original implementation.. I will fix this back.